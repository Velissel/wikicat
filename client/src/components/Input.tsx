import { useField } from 'formik';
import React from 'react';
import { FormControl, FormLabel } from 'react-bootstrap';

interface InputProps {
  label?: string;
  name: string;
}
export default function Input(props: InputProps) {
  const { label, name } = props;
  const [field] = useField(name);

  return (
    <>
      {label && <FormLabel htmlFor={name}>{label}</FormLabel>}
      <FormControl {...field} type="text" id={name} />
    </>
  );
}
