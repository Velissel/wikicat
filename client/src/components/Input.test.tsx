import React from 'react';
import { render, screen } from '@testing-library/react';
import Input from './Input';
import { Formik } from 'formik';
import { Form } from 'react-bootstrap';

describe('Input component', () => {
  it('should render', () => {
    const INIT = { input: '' };
    const { container } = render(
      <Formik initialValues={INIT} onSubmit={jest.fn()}>
        <Form>
          <Input name="input" label="input" />
        </Form>
      </Formik>,
    );
    expect(screen.getByRole('textbox', { name: 'input' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
