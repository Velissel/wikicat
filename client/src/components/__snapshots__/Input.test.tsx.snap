// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render 1`] = `
<div>
  <form
    class=""
  >
    <label
      class="form-label"
      for="input"
    >
      input
    </label>
    <input
      class="form-control"
      id="input"
      name="input"
      type="text"
      value=""
    />
  </form>
</div>
`;
